AK/Vector.h

```diff
@@ -275,22 +277,40 @@ class Vector {
             }
         }
         new (slot(index)) T(forward<U>(value));
+        return true;
+    }
+
+    template<typename U = T>
+    void insert(size_t index, U&& value)
+    {
+        auto did_allocate = try_insert<U>(index, forward<U&&>(value));
+        VERIFY(did_allocate);
     }
 
     template<typename C, typename U = T>
-    void insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
+    [[nodiscard]] bool try_insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
     {
         for (size_t i = first_index; i < size(); ++i) {
             if (callback(at(i))) {
-                insert(i, forward<U>(value));
+                if (!try_insert(i, forward<U>(value)))
+                    return false;
                 if (inserted_index)
                     *inserted_index = i;
-                return;
+                return true;
             }
         }
-        append(forward<U>(value));
+        if (!try_append(forward<U>(value)))
+            return false;
         if (inserted_index)
             *inserted_index = size() - 1;
+        return true;
+    }
+
+    template<typename C, typename U = T>
+    void insert_before_matching(U&& value, C callback, size_t first_index = 0, size_t* inserted_index = nullptr)
+    {
+        auto did_allocate = try_insert_before_matching(forward<U&&>(value), callback, first_index, inserted_index);
```

Comment:
```suggestion
        auto did_allocate = try_insert_before_matching(forward<U>(value), callback, first_index, inserted_index);
```

Reply:
I tried this originally, but encountered some strange template errors. If you give me a few moments, I'll try to get the error messages.

Reply:
Weird. I try compiling before making the suggestion (because I was trying to understand why you did it this way), and it seemed to work.
Please post the error messages if you get any :)

Reply:
Ok, it appears to compile fine now. I originally wrote it as `forward<U>(value)` the first time I tried to compile it, but then I encountered some compiler errors, some of which I managed to fix by changing to `forward<U&&>(value)`. I then fixed some other mistakes in the code and got it to build, so I assumed all of the changes I made were needed. I guess `forward<U>(value)` was correct all along, and all I managed to do was surpress some of the later errors by changing it.

I've changed it to be `forward<U>(value)` again, gonna post a new revision.